modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/metadata/MetadataExtensionFunctionalTestCase.java

```diff
@@ -181,21 +185,37 @@ protected void assertExpectedType(MetadataResult<ParameterMetadataDescriptor> de
     if (!StringUtils.isBlank(name)) {
       assertThat(descriptor.get().getName(), is(name));
     }
+    assertThat(descriptor.get().isDynamic(), is(false));
   }
 
   protected void assertExpectedType(MetadataResult<TypeMetadataDescriptor> descriptor, MetadataType type) throws IOException {
     assertThat(descriptor.get().getType(), is(type));
   }
 
-  protected void assertExpectedType(MetadataResult<ParameterMetadataDescriptor> descriptor, String name, MetadataType type)
+  protected void assertExpectedType(MetadataResult<ParameterMetadataDescriptor> descriptor, String name, MetadataType type,
+                                    boolean isDynamic)
       throws IOException {
+    assertThat(descriptor.isSuccess(), is(true));
     assertThat(descriptor.get().getType(), is(type));
     if (!StringUtils.isBlank(name)) {
       assertThat(descriptor.get().getName(), is(name));
     }
+    assertThat(descriptor.get().isDynamic(), is(isDynamic));
   }
 
   protected Set<MetadataKey> getKeysFromContainer(MetadataKeysContainer metadataKeysContainer) {
     return metadataKeysContainer.getKeys(metadataKeysContainer.getCategories().iterator().next()).get();
   }
+
+  protected void assertSuccess(MetadataResult<?> metadata) {
+    if (!metadata.isSuccess()) {
+      if (metadata.getFailure().isPresent()) {
```

Comment:
if it aint' a success, you should actually assert for the failure to be present rather than an innocent if